modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/headless/batch/engine/resource/v1_0/test/ExportTaskResourceTest.java

```diff
@@ -113,6 +126,99 @@ public void testPostExportTask() throws Exception {
 		}
 	}
 
+	@Test
+	public void testPostExportTaskWithFiltering() throws Exception {
+		try (LogCapture logCapture = LoggerTestUtil.configureLog4JLogger(
+				"com.liferay.batch.engine.internal." +
+					"BatchEngineExportTaskExecutorImpl",
+				LoggerTestUtil.ERROR)) {
+
+			ObjectDefinition objectDefinition =
+				ObjectDefinitionTestUtil.publishObjectDefinition(
+					Collections.singletonList(
+						ObjectFieldUtil.createObjectField(
+							ObjectFieldConstants.BUSINESS_TYPE_TEXT,
+							ObjectFieldConstants.DB_TYPE_STRING,
+							_OBJECT_FIELD_NAME_TEXT)),
+					ObjectDefinitionConstants.SCOPE_COMPANY,
+					TestPropsValues.getUserId());
+
+			String objectDefinitionRESTContextPath =
+				objectDefinition.getRESTContextPath();
+
+			ObjectEntryTestUtil.addObjectEntry(
+				objectDefinition, _OBJECT_FIELD_NAME_TEXT, "TestObject1");
+			ObjectEntryTestUtil.addObjectEntry(
+				objectDefinition, _OBJECT_FIELD_NAME_TEXT, "TestObject2");
+			ObjectEntryTestUtil.addObjectEntry(
+				objectDefinition, _OBJECT_FIELD_NAME_TEXT, "Object3");
+
+			String queryParametersString = StringBundler.concat(
+				"restrictFields=actions&filter=contains%28",
+				_OBJECT_FIELD_NAME_TEXT, "%2C%20%27Test%27%29");
+
+			JSONObject jsonObject = HTTPTestUtil.invokeToJSONObject(
+				null,
+				StringBundler.concat(
+					"headless-batch-engine/v1.0/export-task",
+					"/com.liferay.object.rest.dto.v1_0.ObjectEntry/JSON?",
+					"taskItemDelegateName=", objectDefinition.getName(), "&",
+					queryParametersString),
+				Http.Method.POST);
+
+			String actualExecuteStatus = null;
+
+			while (true) {
```

Comment:
I see a similar code to this one in the `_testPostExportTask` method. Can we reuse it?